library/src/main/java/com/firebase/ui/FirebaseArray.java

```diff
@@ -64,6 +107,24 @@ public DataSnapshot getItem(int index) {
         return mSnapshots.get(index);
     }
 
+    private void setup() {
+        if(mQuery != null) {
+            cleanup();
+        }
+        if(mPageSize == 0) {
+            mQuery = mOriginalQuery;
+        }
+        else if(mOrderASC == true){
+            mQuery = mOriginalQuery.limitToFirst(mCurrentSize);
+        }
+        else {
+            mQuery = mOriginalQuery.limitToLast(mCurrentSize);
+        }
+        setSyncing(true);
+        mQuery.addChildEventListener(this);
+        mQuery.addListenerForSingleValueEvent(this);
```

Comment:
I agree that the simplicity of this approach comes at the cost of bandwidth. Loading pages separately could result in a much more complex system (Managing listeners in order to receive updates for previously loaded pages or using startAt() in conjunction with moveable children in Queries could become issues). I'll have to think about how to solve this properly!